include/SugarFolders/SugarFolders.php

```diff
@@ -695,6 +695,23 @@ public function addBean(SugarBean $bean)
             return false;
         }
 
+        /*
+        Fix issue #9192 - Duplicating rows for folders_rel
+        First check if a row with the same data already exists
+        If so, return false
+        */
+
+        $q = "SELECT id FROM folders_rel WHERE".
+            " folder_id = ".$this->db->quoted($this->id).
+            " AND polymorphic_module = ".$this->db->quoted($bean->module_dir).
+            " AND polymorphic_id = ".$this->db->quoted($bean->id).
+            " AND deleted = 0";
+
+        if($this->db->fetchByAssoc($this->db->query($q))) {
```

Comment:
Hi @MikeyJC ,

Thank you for the pull request.

I was thinking that probably we should also handle errors when running the query. If the `fetchByAssoc` returns `false` that means there was an error running the query. Thus, in that case we should probably also log and return.

What do you think?